src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/ApiIndexFunction.java

```diff
@@ -1,30 +1,59 @@
 package io.stargate.sgv2.jsonapi.service.schema.tables;
 
+import io.stargate.sgv2.jsonapi.config.constants.TableDescConstants;
+import io.stargate.sgv2.jsonapi.exception.SchemaException;
 import io.stargate.sgv2.jsonapi.exception.checked.UnknownCqlIndexFunctionException;
+import java.util.HashMap;
 import java.util.Map;
 
+/**
+ * ApiIndexFunction is a function that is applied in indexes on CQL collection type. <br>
+ * Data API createIndex table command has defaults indexFunction for map/set/list, defaults are
+ * entries(map), values(set), values(list). Data API does not support frozen map/set/list, so FULL
+ * index creation on frozen column will also not be supported.
+ */
 public enum ApiIndexFunction {
   KEYS("keys"),
   VALUES("values"),
   ENTRIES("entries");
 
-  private final String cqlFunction;
+  public final String cqlFunction;
 
-  private static final Map<String, ApiIndexFunction> FUNCTION_MAP =
-      Map.of(
-          KEYS.cqlFunction, KEYS,
-          VALUES.cqlFunction, VALUES,
-          ENTRIES.cqlFunction, ENTRIES);
+  private static final Map<String, ApiIndexFunction> FUNCTION_MAP;
+
+  static {
+    FUNCTION_MAP = new HashMap<>();
+    for (ApiIndexFunction function : ApiIndexFunction.values()) {
+      FUNCTION_MAP.put(function.cqlFunction.toLowerCase(), function);
+    }
+  }
 
   ApiIndexFunction(String cqlFunction) {
     this.cqlFunction = cqlFunction;
   }
 
   public static ApiIndexFunction fromCql(String cqlFunction)
       throws UnknownCqlIndexFunctionException {
-    if (FUNCTION_MAP.containsKey(cqlFunction)) {
-      return FUNCTION_MAP.get(cqlFunction);
+    if (cqlFunction == null || !FUNCTION_MAP.containsKey(cqlFunction.toLowerCase())) {
+      throw new UnknownCqlIndexFunctionException(cqlFunction);
     }
-    throw new UnknownCqlIndexFunctionException(cqlFunction);
+    return FUNCTION_MAP.get(cqlFunction.toLowerCase());
+  }
+
+  public static ApiIndexFunction fromDollarCommand(String indexFunctionWithDollarSign) {
```

Comment:
-1 the API names for things should be in the api packages I think not in our core schema here. 

and we should do this as case insensitive